include/wx/string.h

```diff
@@ -2132,6 +2132,9 @@ class WXDLLIMPEXP_BASE wxString
       // insert a double into string
   wxString& operator<<(double d)
     { return (*this) << Format(wxT("%g"), d); }
+  // insert a pointer as hex into string
+  wxString &operator<<(void* p)
+    { return (*this) << Format(wxS("%p"), p); }
```

Comment:
Sorry, I don't like this, this makes it far too simple to print something as a pointer unintentionally. I suspect that having these `operator<<()`s in wxString was a mistake anyhow, but it was very tempting for the common arithmetic types because you often need to construct strings with them. Printing out pointers is much more rarely needed so IMO there is nothing wrong with just using `s += wxString::Format("%p", p);` for them explicitly.

Reply:
Printing a void pointer as a hex representation is also part of the C++ standard, so it's not unexpected behavior. Also I find it rather uninituitive that you can use the stream operator, just like any programmer is used to it from the standard, and then some parts are missing for no obvious reason.
The stream operators exist, so why should they not be used in the proper way?
IMO it looks rather clumsy if I have to write something like:
`str << "Window: " << wxString::Format("%p", window)`
instead of 
`str << "Window: " << window`

I also don't see why it should be printed unintentionally, because if it is a pointer, then it has to be dereferenced to be used in the stream in order to print it's content. You can't really do
`str << "Window: " << window`
and expect to have something else printed, right?

Also I prefer the typesafety that the stream operator provides as opposed to the printf style formatting, which is potentially much more unsafe and requires manual updating if the types are changed for some reason.

Reply:
The reason is that some objects are convertible to a pointer, sometimes without really being a pointer at all, and printing them as such would be confusing/unexpected. Adding a new overload could also break compilation of the existing code if there is now an ambiguity between converting to int or a pointer. And even if this is unlikely (which doesn't mean that it won't happen, just that it will happen in very few cases), I honestly just don't want to deal with the possible fallout from this for what seems like very little gain.

Personally I'd rather avoid adding anything to `wxString` at all, this class made sense back in 1995, but nowadays there is really not much reason to use it for anything and, in particular, using fmt library for formatting is much, much better.